### ISSUE TITLE
Python 3.14: AttributeError: module 'ast' has no attribute 'Str'

### ISSUE THREAD

**Issue by ondrejj** [malthe/chameleon#430]
According to: https://docs.python.org/dev/whatsnew/3.14.html#id2

Remove the following classes. They were all deprecated since Python 3.8, and
have emitted deprecation warnings since Python 3.12:
ast.Bytes
ast.Ellipsis
ast.NameConstant
ast.Num
ast.Str

Use ast.Constant instead. As a consequence of these removals, user-defined
visit_Num, visit_Str, visit_Bytes, visit_NameConstant and visit_Ellipsis
methods on custom ast.NodeVisitor subclasses will no longer be called when the
NodeVisitor subclass is visiting an AST. Define a visit_Constant method
instead.

https://bugzilla.redhat.com/show_bug.cgi?id=2322715

**Comment by diegorusso** [comment]
I confirm I have encountered this error while running pyperformance with CPython main.

**Comment by malthe** [comment]
Just noting that the final release for 3.14 is October, 2025: https://peps.python.org/pep-0745/.

But time moves fast, so it would be good to be on top of this soonish ...

**Comment by diegorusso** [comment]
> Just noting that the final release for 3.14 is October, 2025: https://peps.python.org/pep-0745/.
> 
> But time moves fast, so it would be good to be on top of this soonish ...

That's correct. I guess from the user point of view there is time, but from the CPython development point of view this causes some issues. Chameleon is used in pyperformance (https://github.com/python/pyperformance/tree/main/pyperformance/data-files/benchmarks/bm_chameleon) and currently it is failing. `main` branch is mainly where we track our performances and it would be good to have it working again :)
I'll be monitoring this issue and as soon as there is a new release with the fix, I'll update pyperformance.

Thanks!

**Comment by fschulze** [comment]
The fix for this was merged today.

**Comment by stevepiercy** [comment]
This issue can be closed. See:

- https://github.com/malthe/chameleon/releases/tag/4.6.0
- https://github.com/malthe/chameleon/pull/434

**Comment by fsc-eriker** [comment]
FWIW the very first link is no longer correct; the location for these notes is now https://docs.python.org/dev/whatsnew/3.14.html#id9
